src/figuresClasses.ts

```diff
@@ -1,11 +1,83 @@
-export interface Figure {}
+type Shapes = 'triangle' | 'circle' | 'rectangle';
+type Colors = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shapes;
+  color: Colors;
+  getArea(): number;
+}
+
+function checkError(condition: boolean, message: string): void {
+  if (condition) {
+    throw new Error(message);
+  }
+}
+
+export class Triangle implements Figure {
+  readonly shape = 'triangle';
+
+  constructor(
+    public color: Colors,
+    private a: number,
+    private b: number,
+    private c: number,
```

Comment:
You don't need to manually assign the constructor parameters to class properties when using access modifiers in TypeScript. The compiler will do this for you.